src/benchmark.cc

```diff
@@ -623,7 +623,7 @@ void RunBenchmark(const benchmark::internal::Benchmark::Instance& b,
 
   for (int i = 0; i < FLAGS_benchmark_repetitions; i++) {
     std::string mem;
-    while (true) {
+    for (;;) {
```

Comment:
Yes, with W4 it does... (warning C4127: conditional expression is constant)
An alternative would be to disable this warning in the CMake file.